src/loader.py

```diff
@@ -220,51 +242,72 @@ def merge_summary_documents(summaries, metadata_list):
 
 
 def get_file_summary(path: str):
-    client = Groq(
-        api_key=os.environ.get("GROQ_API_KEY"),
-    )
+    # client = Groq(
+    #     api_key=os.environ.get("GROQ_API_KEY"),
+    # )
     reader = SimpleDirectoryReader(input_files=[path]).iter_data()
 
     docs = next(reader)
     splitter = TokenTextSplitter(chunk_size=6144)
     text = splitter.split_text("\n".join([d.text for d in docs]))[0]
     doc = Document(text=text, metadata=docs[0].metadata)
-    summary = dispatch_summarize_document_sync(doc, client)
+    summary = dispatch_summarize_document_sync(doc)
     return summary
 
 
-def dispatch_summarize_document_sync(doc, client):
+def dispatch_summarize_document_sync(doc):
     if isinstance(doc, ImageDocument):
-        return summarize_image_document_sync(doc, client)
+        return summarize_image_document_sync(doc)
     elif isinstance(doc, Document):
-        return summarize_document_sync({"content": doc.text, **doc.metadata}, client)
+        return summarize_document_sync({"content": doc.text, **doc.metadata})
     else:
         raise ValueError("Document type not supported")
 
 
-def summarize_document_sync(doc, client):
+def summarize_document_sync(doc):
     PROMPT = """
 You will be provided with the contents of a file along with its metadata. Provide a summary of the contents. The purpose of the summary is to organize files based on their content. To this end provide a concise but informative summary. Make the summary as specific to the file as possible.
 
 Write your response a JSON object with the following schema:
-    
-```json 
+
+```json
 {
     "file_path": "path to the file including name",
     "summary": "summary of the content"
 }
 ```
 """.strip()
 
-    chat_completion = client.chat.completions.create(
-        messages=[
-            {"role": "system", "content": PROMPT},
-            {"role": "user", "content": json.dumps(doc)},
+    client = ollama.Client()
+    chat_completion = client.chat(
+        messages = [
+            {
+                "role": "system",
+                "content": PROMPT},
+            {
+                "role": "user",
+                "content": json.dumps(doc),
+            },
```

Comment:
The Ollama SDK might not be the greatest tool to use. I'd strongly recommend using LiteLLM as the replacement and then setting the model to `ollama/<model>`

https://github.com/BerriAI/litellm
https://docs.litellm.ai/docs/providers/ollama

That said, LiteLLM has not updated the model registry to include llama 3.2.

Reply:
I will definitely look into this as I was unaware of it before.

I won't have time to work on this anytime soon, however.